src/hotspot/share/runtime/vmThread.hpp

```diff
@@ -103,9 +103,11 @@ class VMThread: public NamedThread {
     assert(Thread::current()->is_VM_thread(), "Must be");
     return _cur_vm_operation;
   }
+
   static VM_Operation::VMOp_Type vm_op_type()     {
-    assert(Thread::current()->is_VM_thread(), "Must be");
-    return _cur_vm_operation->type();
+    VM_Operation* op = vm_operation();
```

Comment:
Why use vm_operation() instead of direct access to _cur_vm_operation? Elsewhere you made the reverse change.

Reply:
It was suggested by @shipilev todo that, so we don't need the:
`assert(Thread::current()->is_VM_thread(), "Must be");`
In that method also.

In the other cases we already had access to _cur_vm_operation and it was confusing to mix.